packages/medusa-payment-stripe/src/api/utils/utils.ts

```diff
@@ -0,0 +1,254 @@
+import { AwilixContainer } from "awilix"
+import Stripe from "stripe"
+import {
+  AbstractCartCompletionStrategy,
+  CartService,
+  IdempotencyKeyService,
+  PostgresError,
+} from "@medusajs/medusa"
+import { EOL } from "os"
+import { MedusaError } from "medusa-core-utils"
+
+const PAYMENT_PROVIDER_KEY = "pp_stripe"
+
+export function constructWebhook({
+  signature,
+  body,
+  container,
+}: {
+  signature: string | string[] | undefined
+  body: any
+  container: AwilixContainer
+}): Stripe.Event {
+  const stripeProviderService = container.resolve(PAYMENT_PROVIDER_KEY)
+  return stripeProviderService.constructWebhookEvent(body, signature)
+}
+
+export function isPaymentCollection(id) {
+  return id && id.startsWith("paycol")
+}
+
+export function buildError(event: string, err: Stripe.StripeRawError): string {
+  let message = `Stripe webhook ${event} handling failed\n${
+    err?.detail ?? err?.message
+  }`
+  if (err?.code === PostgresError.SERIALIZATION_FAILURE) {
+    message = `Stripe webhook ${event} handle failed. This can happen when this webhook is triggered during a cart completion and can be ignored. This event should be retried automatically.\n${
+      err?.detail ?? err?.message
+    }`
+  }
+  if (err?.code === "409") {
+    message = `Stripe webhook ${event} handle failed.\n${
+      err?.detail ?? err?.message
+    }`
+  }
+
+  return message
+}
+
+export async function handlePaymentHook(event, req, res, paymentIntent) {
+  const logger = req.scope.resolve("logger")
+
+  const cartId = paymentIntent.metadata.cart_id // Backward compatibility
+  const resourceId = paymentIntent.metadata.resource_id
+
+  switch (event.type) {
+    case "payment_intent.succeeded":
+      try {
+        await onPaymentIntentSucceeded({
```

Comment:
The issue that makes us to do this is that with some provider the amount capturable does not get fired because of how it works and only succeeded is fired (blik, ideal...)

Reply:
But we will have to test that one way or the other, I have tested stripe but I don't have all of them

Reply:
> some providers the amount capturable does not get fired

That's right! I forgot that minor, but important detail 😅